### PR TITLE
Add AKS review app terraform

### PR DESCRIPTION
**Pull request by RMcVelia**
### Trello card

https://trello.com/c/lTSOYCTM/589-deploy-application

### Context

Add review apps for AKS

### Changes proposed in this pull request

Add code for AKS review app deployment
- terraform code
- Makefile changes
- change rails database.yml to use DATABASE_URL for aks
- exclude healthcheck from ssl

### Guidance to review

- check the existing paas review app for this PR is working as expected
- make review_aks terraform-apply-aks PR_NUMBER=n IMAGE_TAG=sha-2e9c219

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -21,9 +21,17 @@ test:
   port: <%= ENV['DB_PORT'] %>
 
 production: &production
+  <% if ENV.key?("VCAP_APPLICATION") %>
+
   <<: *default
   url: <%= Rails.application.config.x.vcap_services.dig("postgres", 0, "credentials", "uri") %>
 
+  <% else %>
+  <<: *default
```
Comment (saliceti) [inline config/database.yml:30]: Can this line be moved outside of the if so it's not repeated?

File: terraform/aks/application.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+module "web_application" {
+  source = "./vendor/modules/aks//aks/application"
+
+  is_web = true
+
+  namespace    = var.namespace
+  environment  = local.environment
+  service_name = var.service_name
+  probe_path   = "/healthcheck.json"
```
Comment (saliceti) [inline terraform/aks/application.tf:42]: The healthcheck.json may be too heavy for the probe as it checks all the dependencies, as per: https://github.com/DFE-Digital/get-into-teaching-app/blob/4aa365ce19dd09d6da9c08ba14e52ec9d4abac06/app/models/healthcheck.rb#L4
Can we replace it with a simple one like: https://github.com/DFE-Digital/teacher-pay-calculator/blob/main/config/routes.rb#L11

File: terraform/aks/database.tf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  config_short                = var.config_short
+  cluster_configuration_map   = module.cluster_data.configuration_map
+  use_azure                   = var.deploy_azure_backing_services
+  azure_enable_monitoring     = var.enable_monitoring
+  azure_enable_backup_storage = var.enable_postgres_backup_storage
+  server_version              = "14"
+}
+
+
+module "redis-cache" {
```
Comment (saliceti) [inline terraform/aks/database.tf:18]: Can we pin server_version to 6?

File: terraform/aks/output.tf [hunk 4]
```diff
@@ -0,0 +1,3 @@
+output "url" {
+  value = "https://${module.web_application.hostname}/"
```
Comment (saliceti) [inline terraform/aks/output.tf:2]: The `url` output may be used here for simplicity (https://github.com/DFE-Digital/terraform-modules/blob/main/aks/application/outputs.tf#L5)

File: terraform/aks/database.tf (hunk 3 above)
Comment (RMcVelia) [reply terraform/aks/database.tf:18]: default is 6 in the module?
https://github.com/DFE-Digital/terraform-modules/blob/main/aks/redis/variables.tf#L49

File: terraform/aks/output.tf (hunk 4 above)
Comment (RMcVelia) [reply terraform/aks/output.tf:2]: The output.tf is created by the new service template. Not sure what to change here?

File: terraform/aks/database.tf (hunk 3 above)
Comment (saliceti) [reply terraform/aks/database.tf:18]: Yes but we may want to default to 7 in the module once it's available. So it should be pinned in GIT so GIT it is not updated automatically.

File: terraform/aks/output.tf (hunk 4 above)
Comment (RMcVelia) [reply terraform/aks/output.tf]: updated

File: terraform/aks/database.tf (hunk 3 above)
Comment (RMcVelia) [reply terraform/aks/database.tf:18]: pinned to 6

**Review by saliceti — DISMISSED** [review]
🙏

**Review by saliceti — APPROVED** [review]

**Pull request merged**